mentoring/components/mrq.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014 edX
+#
+# Authors:
+#          Xavier Antoviaque <xavier@antoviaque.org>
+#
+# This software's license gives you freedom; you can copy, convey,
+# propagate, redistribute and/or modify this program under the terms of
+# the GNU Affero General Public License (AGPL) as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Imports ###########################################################
+
+import logging
+
+from xblock.fields import List, Scope, Boolean
+from xblock.validation import ValidationMessage
+from .questionnaire import QuestionnaireAbstractBlock
+from xblockutils.resources import ResourceLoader
+
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+
+
+# Classes ###########################################################
+
+class MRQBlock(QuestionnaireAbstractBlock):
+    """
+    An XBlock used to ask multiple-response questions
+    """
+    student_choices = List(help="Last submissions by the student", default=[], scope=Scope.user_state)
+    required_choices = List(
+        display_name="Required Choices",
+        help=("Specify the value[s] that students must select for this MRQ to be considered correct. "),
+        scope=Scope.content,
+        list_values_provider=QuestionnaireAbstractBlock.choice_values_provider,
+        list_style='set',  # Underered, unique items. Affects the UI editor.
+        default=[],
+    )
+    ignored_choices = List(
+        display_name="Ignored Choices",
+        help=(
+            "Specify the value[s] that are neither correct nor incorrect. "
+            "Any values not listed as required or ignored will be considered wrong."
+        ),
+        scope=Scope.content,
+        list_values_provider=QuestionnaireAbstractBlock.choice_values_provider,
+        list_style='set',  # Underered, unique items. Affects the UI editor.
+        default=[],
+    )
+    hide_results = Boolean(display_name="Hide results", scope=Scope.content, default=False)
+    editable_fields = ('question', 'required_choices', 'ignored_choices', 'message', 'weight', 'hide_results', )
```

Comment:
I am currently unable to edit new MRQs, it asks for a unique ID, but I'm not able to set one - maybe because the fields don't use `QuestionnaireAbstractBlock.editable_fields` like for the MCQs?

![screenshot-sandbox opencraft com 18010 2015-03-02 15-34-17](https://cloud.githubusercontent.com/assets/514483/6442710/066cee0a-c0f2-11e4-8ce5-19fb4beafecf.png)